controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -168,6 +169,20 @@ func GetIntegrationPipelineRunStatus(adapterClient client.Client, ctx context.Co
 		}
 	}
 
+	taskRuns, err := a.loader.GetAllTaskRunsWithMatchingPipelineLabel(adapterClient, ctx, pipelineRun)
+	if err != nil {
```

Comment:
I'm maybe too much of rebel, how about just failing without retries?


```diff
diff --git a/controllers/integrationpipeline/integrationpipeline_adapter.go b/controllers/integrationpipeline/integrationpipeline_adapter.go
index 04a2fcf..2ba9032 100644
--- a/controllers/integrationpipeline/integrationpipeline_adapter.go
+++ b/controllers/integrationpipeline/integrationpipeline_adapter.go
@@ -170,7 +170,7 @@ func GetIntegrationPipelineRunStatus(adapterClient client.Client, ctx context.Co
 
        outcome, err := h.GetIntegrationPipelineRunOutcome(adapterClient, ctx, pipelineRun)
        if err != nil {
-               return intgteststat.IntegrationTestStatusTestFail, "", fmt.Errorf("failed to evaluate inegration test results: %w", err)
+               return intgteststat.IntegrationTestStatusTestInvalid, fmt.Sprintf("failed to evaluate inegration test results: %s", err), nil
        }
 
        if !outcome.HasPipelineRunPassedTesting() {
```

Reply:
I agree with you that you're too much of a rebel

Reply:
This is interesting, thanks for the suggestion! I'm now in the process of deploying this suggested change on a cluster to see how it looks like and then post here for comparison to decide which one to go with.

Reply:
So, this is how the status looks like as part of the current logic within this PR:
![image](https://github.com/redhat-appstudio/integration-service/assets/43273420/333e8656-98fc-4c33-8b89-ce7281571e7d)

Reply:
And this is how the status looks like as part of the changes suggested within this thread:
![image](https://github.com/redhat-appstudio/integration-service/assets/43273420/5b61a614-2aee-4abf-9ede-84e974f4834f)

Reply:
Honestly, both of these messages look descriptive enough to clearly explain why the status is marked as Invalid. Compared to Approach 2, Approach 1's implementation makes some extra calls and duplicates some logic. I'm leaning towards the 2nd approach. What do you all think?

Reply:
the main issue is, that the second approach won't retry in any case

Reply:
Oh right, hmm that's not good...

Reply:
Generally, not retrying for all errors, doesn't sound good.
Looking at the current logic, we would miss out on retrying [this error](https://github.com/redhat-appstudio/integration-service/blob/main/helpers/integration.go#L161-L165):
```
error while compiling json data for schema validation ....
```

Reply:
yes, but wrong example, that's unrecoverable error that would need code change, so retrying won't solve anything in that case

I'm fine with merging it as is

Reply:
+1 on merging as is as discussed on standup.